golden_hammer/module.go

```diff
@@ -0,0 +1,275 @@
+}
+
+func (h *GoldenHammer) hammerTime() {
+	ticker := time.NewTicker(h.config.Interval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-h.ctx.Done():
+			return
+		case <-ticker.C:
+			err := h.registerServiceBaseURLs()
+			if err != nil {
+				log.Logger().WithError(err).Warn("Auto-fix error")
+			}
+		}
+	}
+}
+
+// registerServiceBaseURLs registers the node's services HTTP base URL on its DIDs.
+// This base URL is used to discover HTTP services (for now only OpenID4VCI wallet/issuer metadata).
+//   - Make a list of owned DIDs
+//   - Filter documents without node-http-services-base-url service
+//   - Sort: nodes without NutsComm service go first (so "vendor DID documents" go first).
+//   - Try to resolve metadata (TLSIdentifierResolver), if successful, continue
+//   - No NutsComm service reference? Register node-http-services-base-url service it on the DID document itself.
+//   - Has NutsComm service reference? Register node-http-services-base-url reference to ref'd DID document, if present there.
+func (h *GoldenHammer) registerServiceBaseURLs() error {
+	documents, err := h.listDocumentToFix()
+	if err != nil {
+		return err
+	}
+	for _, document := range documents {
+		var endpointToRegister url.URL
+		serviceEndpoint := getServiceEndpoint(document, transport.NutsCommServiceType)
+		if didservice.IsServiceReference(serviceEndpoint) {
+			// Care organization DID document, register service pointing to vendor DID.
+			parentDID, err := didservice.GetDIDFromURL(serviceEndpoint)
+			if err != nil {
+				// Invalid NutsComm reference, skip
+				log.Logger().WithError(err).
+					Debugf("Invalid NutsComm reference on DID %s: %s", document.ID, serviceEndpoint)
+				continue
+			}
+			// Only if the referenced document actually contains the service
+			if h.resolveContainsService(parentDID, types.BaseURLServiceType) {
+				log.Logger().Debugf("Could not resolve '%s' service in referenced (NutsComm) DID document (did=%s), skipping fix for DID: %s", types.BaseURLServiceType, parentDID.String(), document.ID)
+				continue
+			}
+			// All us OK
+			endpointToRegister = didservice.MakeServiceReference(parentDID, types.BaseURLServiceType).URL
+		} else {
+			// Vendor DID document, register resolved identifier
+			identifier, err := h.tryResolveURL(document.ID)
+			if identifier == nil || err != nil {
+				log.Logger().WithError(err).
+					Debugf("Could not resolve vendor's services base URL for DID %s, skipping fix", document.ID)
+				continue
+			}
+			endpointToRegister = *identifier
+		}
+		_, err := h.didmanAPI.AddEndpoint(h.ctx, document.ID, types.BaseURLServiceType, endpointToRegister)
+		if err != nil {
+			log.Logger().WithError(err).
+				Errorf("Unable to register DID services base URL (did=%s): %s", document.ID, endpointToRegister.String())
+		} else {
+			h.fixedDocumentDIDs[document.ID.String()] = true
+			log.Logger().Infof("Registered DIDs services base URL (did=%s): %s", document.ID, endpointToRegister.String())
+		}
```

Comment:
log mania on first run for vendors with many DIDs. didman also logs twice (1 debug, 1 info) that it added an endpoint.

Reply:
and an audit log as well? So debug is enough

Reply:
Will just add a single log message when it has fixed anything